src/domain/WinningDetail.js

```diff
@@ -0,0 +1,73 @@
+import Lotto from "./Lotto.js";
+import PurchaseHistory from "./PurchaseHistory.js";
+
+class WinningDetail {
+  #winner;
+
+  // 객체를 주입받아서 사용한다.
+  constructor({ purchaseHistory, lotto }) {
+    if ((!purchaseHistory) instanceof PurchaseHistory) {
+      throw new Error("당첨 내역을 출력할 수 없어요.");
+    }
```

Comment:
아래처럼 작성하지 않고 현재 코드처럼 작성하신 이유가 궁금하네요 !

```suggestion
    if ((!purchaseHistory instanceof PurchaseHistory)) {
      throw new Error("당첨 내역을 출력할 수 없어요.");
    }
```

Reply:
앗 이 부분은 오타입니다. 느낌표를 쓰는 것을 쓰는 좋아하지 않지만, 급하게 구현했던 부분이 남아있는 거 같습니다 
=== false로 변경해서 반영하겠습니다!

Reply:
오 요것도 재밌는 부분이네요 `!` 를 쓰는 것을 좋아하지 않으시는 이유가 궁금합니다 ㅎㅎ 

저는 개인적으로는 nullable 한 값을 검증할 때 `!`를 사용해서 `falsy`한 값들을 모두 잡아주는 걸 선호해서 어떤 배경으로 안 좋아하시는지 궁금하네용 !

Reply:
음.. 약간 일반적인 코딩 컨벤션을 깨뜨린다고 생각했습니다! 명시적으로 false를 써주지 않으니 나중에 오타가 발생했을 때, 쉽게 찾아서 알아보기가 어려운 경우가 있더라구요, 제 개인적인 디버깅 경험에서 나온 이야기입니다.
이 부분은 제 개인적인 생각입니다! 팀에서 정해진 ground rule이 ! 라면, 그것을 따르려고 합니다. (항상 제가 맞다고는 생각하지 않아서요 ㅎㅎ)

Reply:
아하 그렇군요 경험 공유해주셔서 감사합니다 호정님 충분히 이해가 되는 상황이네요 ㅎㅎ 👍 👍 👍